python/revng/api/_capi.py

```diff
@@ -163,12 +169,18 @@ def initialize(
     """Initialize library, must be called exactly once"""
 
     if libraries is None:
-        libraries, _ = collect_libraries(ROOT)
+        if not revng_options_has("load"):
+            libraries, _ = collect_libraries(ROOT)
+        else:
+            libraries = []
     else:
         libraries = to_iterable(libraries)
 
     if pipelines is None:
-        pipelines = collect_files(ROOT, ["share", "revng", "pipelines"], "*.yml")
+        if not revng_options_has("pipeline-path"):
```

Comment:
Otherwise `initialize` would be passing files that (possibly the user) does not expect
e.g.
```
REVNG_OPTIONS="-pipeline-path=foo.yml" osh revng daemon
```

Should not call `collect_pipelines`

Reply:
So you're saying that if in `REVNG_OPTIONS` there's `-pipeline-path` we do not autocollect them?

I think we should either have the C API ignore `REVNG_OPTIONS` entirely or have an argument to enable/disable auto-collection of things. Parsing `REVNG_OPTIONS` like this is sad.

Reply:
What specific situation are we trying to handle here?
Under which circumstances do we have `REVNG_OPTIONS` with `pipeline-path` and we'd like not to load any other pipeline?

Reply:
Added code that removes `REVNG_OPTIONS` when using `revng.api._capi`